core/object/make_virtuals.py

```diff
@@ -0,0 +1,151 @@
+
+
+proto = """
+#define GDVIRTUAL$VER($RET m_name $ARG) \\
+GDNativeExtensionClassCallVirtual _gdvirtual_##m_name = (_get_extension() && _get_extension()->get_virtual) ? _get_extension()->get_virtual(_get_extension()->class_userdata, #m_name) : (GDNativeExtensionClassCallVirtual) nullptr;\\
+StringName _gdvirtual_##m_name##_sn = #m_name;\\
+bool _gdvirtual_##m_name##_call($CALLARGS) $CONST { \\
+	ScriptInstance *script_instance = ((Object*)(this))->get_script_instance();\\
+	if (script_instance) {\\
+		Callable::CallError ce; \\
+		$CALLSIARGS\\
+		$CALLSIBEGINscript_instance->call(_gdvirtual_##m_name##_sn, $CALLSIARGPASS, ce);\\
+		if (ce.error == Callable::CallError::CALL_OK) {\\
+			$CALLSIRET\\
+			return true;\\
+		}    \\
+	}\\
+	if (_gdvirtual_##m_name) {\\
+		$CALLPTRARGS\\
+		$CALLPTRRETDEF\\
+		_gdvirtual_##m_name(_get_extension_instance(),$CALLPTRARGPASS,$CALLPTRRETPASS);\\
+		$CALLPTRRET\\
+		return true;\\
+	}\\
+\\
+    return false;\\
+}\\
+\\
+_FORCE_INLINE_ static MethodInfo _gdvirtual_##m_name##_get_method_info() { \\
+    MethodInfo method_info;\\
+    method_info.name = #m_name;\\
+    method_info.flags = METHOD_FLAG_VIRTUAL;\\
+    $FILL_METHOD_INFO\\
+    return method_info;\\
+}
+
+
+"""
+
+def generate_version(argcount,const = False, returns = False):
+    s = proto
+    sproto = str(argcount)
+    method_info=""
+    if (returns):
+        sproto +="R"
+        s = s.replace("$RET","m_ret, ")
+        s = s.replace("$CALLPTRRETDEF","PtrToArg<m_ret>::EncodeT ret;")
+        method_info += "\tmethod_info.return_val = GetTypeInfo<m_ret>::get_class_info();\\\n"
+    else:
+        s = s.replace("$RET","")
+        s = s.replace("$CALLPTRRETDEF","")
+        
+    if (const):
+        sproto +="C"
+        s = s.replace("$CONST","const")
+        method_info += "\tmethod_info.flags|=METHOD_FLAG_CONST;\\\n"
+    else:
+        s = s.replace("$CONST","")
+        
+    s = s.replace("$VER",sproto)
+    argtext = ""
+    callargtext =""
+    callsiargs = ""
+    callsiargptrs =""
+    callptrargsptr =""
+    if (argcount>0):
+        argtext+=", "
+        callsiargs="Variant vargs["+str(argcount)+"]={"
```

Comment:
consider using fstring:
```python
callsiargs = f"Variant vargs[{argcount}]={{"
```
(note the final `{{` to escape the real `{` character)